app/src/main/java/org/systers/mentorship/utils/PreferenceUtils.kt

```diff
@@ -0,0 +1,128 @@
+package org.systers.mentorship.utils
+
+import android.preference.PreferenceManager
+import org.systers.mentorship.MentorshipApplication
+import org.systers.mentorship.models.User
+
+object PreferenceUtils {
+
+    private val pm = PreferenceManager.getDefaultSharedPreferences(MentorshipApplication.instance)
+
+    private const val ID = "id"
+    private const val USERNAME = "username"
+    private const val NAME = "name"
+    private const val EMAIL = "email"
+    private const val BIO = "bio"
+    private const val LOCATION = "location"
+    private const val OCCUPATION = "occupation"
+    private const val ORGANIZATION = "organization"
+    private const val INTERESTS = "interests"
+    private const val SKILLS = "skills"
+    private const val NEED_MENTORING = "need_mentoring"
+    private const val AVAILABLE_TO_MENTOR = "available_to_mentor"
+    private const val SLACK_USERNAME = "slack_username"
+
```

Comment:
@Priyansh-Kedia This all can be moved to a `companion object`. Doing so will make the file look better and remove the clutter.
As it is always a good practice to separate the constants to a different file and since we have the beautiful Kt. then we can make use of its `companion object`.

Reply:
> @Priyansh-Kedia This all can be moved to a `companion object`. Doing so will make the file look better and remove the clutter.
> As it is always a good practice to separate the constants to a different file and since we have the beautiful Kt. then we can make use of its `companion object`.

I have not added them to companion object as we do not need these constants to be static and adding them in companion would be of no use

Reply:
Can you please elaborate whatever you just quote

Reply:
The thing is 
That we do not need these constants to be static
Static members are part of class and thus remain in memory till application terminates and can’t be ever garbage collected.
Hence we should not be making them static until and unless there is a need

Reply:
The constants don't become static until or unless their wrapping method/api is marked as `@JvmStatic`.
Please See:- [CompanionObjects](https://kotlinlang.org/docs/tutorials/kotlin-for-py/objects-and-companion-objects.html#companion-objects)

Reply:
I am not sure if that is true
Because we can use any function or variable inside the companion object from other class without making an instance and also without using @JvmStatic

Reply:
Don't blindly punch in whatever, just go through the references attached once, you'll get know the api how kt works!

Reply:
I did not blindly say it 
I apologise if it came out that way

@JvmStatic needs to be used if we are using the function/ variable with Java
The difference being

companion object {
       fun demoFun() = 1
}

Now if I do not use @JvmStatic, then calling this method from Java code would require me to use
ClassName.Companion.demoFun()

But as the code in this project is purely in Kotlin, there is no need to use that
I hope I made myself clear